examples/rsa-example.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <iostream>
+#include <mococrw/asymmetric_crypto_ctx.h>
+#include <mococrw/padding_mode.h>
+
+using namespace mococrw;
+
+std::vector<uint8_t> encrypt_rsa(const std::shared_ptr<AsymmetricPublicKey> &pubKey,
+                                 const std::shared_ptr<RSAEncryptionPadding> &padding,
+                                 const std::vector<uint8_t> &data)
+{
+    RSAEncryptionPublicKeyCtx rsaPubCtx(*pubKey, padding);
+
+    std::vector<uint8_t> ciphertext;
+    try {
+        ciphertext = rsaPubCtx.encrypt(data);
+    } catch (const openssl::OpenSSLException &e) {
+        std::cerr << "Failure encrypting data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (const mococrw::MoCOCrWException &e) {
+        std::cerr << "Failure in crypto engine: ";
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return ciphertext;
+}
+
+std::vector<uint8_t> decrypt_rsa(const std::shared_ptr<AsymmetricPrivateKey> &privKey,
```

Comment:
consistency with signature example: Below we handle keys as const ref to the key object.